### DISCUSSION TITLE
Does tokio Tcp have an API or connection disconnection event to check if the connection is disconnected

### DISCUSSION THREAD

**Discussion by Hans-Wu-cn** [tokio-rs/tokio#6285] [Q&A] [answered]
Does tokio Tcp have an API or connection disconnection event to check if the connection is disconnected

**Comment by Darksonn** [comment, marked as answer]
To do that, you should read from the socket. When the connection is closed, you will get a read of length zero.

See #3467 for discussion about waiting for disconnect without reading. It's not possible right now.
